### ISSUE TITLE
[docs] add typings for `__TAURI_ISOLATION_HOOK__`.

### ISSUE THREAD

**Issue by pauliesnug** [tauri-apps/tauri#10459]
Currently it's suggested to use untyped JavaScript for implementing the `__TAURI_ISOLATION_HOOK__`.
Recently I tried to implement the hook in TypeScript with Vite, but I recieve type errors. There doesn't seem to be official typings for `__TAURI_ISOLATION_HOOK__`. What is the general structure of an isolation payload? Does it change often, or is it always the same structure?

**Comment by alfred-liu96** [comment]
Yes, I have encountered the same situation. It seems that currently using JavaScript instead of TypeScript for `__TAURI_ISOLATION_HOOK__` works.
